eclair-core/src/main/scala/fr/acinq/eclair/swap/SwapHelpers.scala

```diff
@@ -0,0 +1,120 @@
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.reflect.ClassTag
+import scala.util.{Failure, Success}
+
+object SwapHelpers {
+
+  def receiveSwapMessage[B <: SwapCommand : ClassTag](context: ActorContext[SwapCommand], stateName: String)(f: B => Behavior[SwapCommand]): Behavior[SwapCommand] =
+    Behaviors.receiveMessage {
+      case m: B => f(m)
+      case m =>
+        context.log.error(s"received unhandled message while at state $stateName of ${m.getClass.getSimpleName}")
+        Behaviors.same
+    }
+
+  def watchForTxConfirmation(replyTo: ActorRef[ZmqWatcher.WatchTxConfirmedTriggered], txId: ByteVector32, minDepth: Long)(implicit watcher: ActorRef[ZmqWatcher.Command]): Unit =
+    watcher ! WatchTxConfirmed(replyTo, txId, minDepth)
+
+  def queryChannelInfo()(implicit context: ActorContext[SwapCommand], register: actor.ActorRef, shortChannelId: ShortChannelId): Unit =
+    register ! Register.ForwardShortId(getInfoAdapter(context).toClassic, shortChannelId, CMD_GET_CHANNEL_INFO(getInfoAdapter(context).toClassic))
+
+  def getInfoAdapter(context: ActorContext[SwapCommand]): ActorRef[CommandSuccess[CMD_GET_CHANNEL_INFO]] = context.messageAdapter[CommandSuccess[CMD_GET_CHANNEL_INFO]](ChannelInfoResponse)
+
+  def queryOnChainBalance()(implicit context: ActorContext[SwapCommand], wallet: OnChainWallet): Unit =
+    context.pipeToSelf(wallet.onChainBalance())(QueryOnChainBalance)
+
+  def payInvoice(paymentInitiator: actor.ActorRef, swapId: String, invoice: Bolt11Invoice)(implicit nodeParams: NodeParams): Unit =
+    paymentInitiator ! SendPaymentToNode(invoice.amount_opt.get, invoice, nodeParams.maxPaymentAttempts, Some(swapId), assistedRoutes = invoice.routingInfo, nodeParams.routerConf.pathFindingExperimentConf.getRandomConf().getDefaultRouteParams, blockUntilComplete = true)
+
+  def watchForPayment(watch: Boolean)(implicit context: ActorContext[SwapCommand]): Unit =
+    if (watch) context.system.classicSystem.eventStream.subscribe(paymentEventAdapter(context).toClassic, classOf[PaymentEvent])
+    else context.system.classicSystem.eventStream.unsubscribe(paymentEventAdapter(context).toClassic, classOf[PaymentEvent])
+
+  def paymentEventAdapter(context: ActorContext[SwapCommand]): ActorRef[PaymentEvent] = context.messageAdapter[PaymentEvent](PaymentEventReceived)
+
+  def send(message: HasSwapId)(implicit context: ActorContext[SwapCommand], register: actor.ActorRef, shortChannelId: ShortChannelId): Unit =
+    register ! Register.ForwardShortId(forwardShortIdAdapter(context).toClassic, shortChannelId, message)
+
+  def forwardShortIdAdapter(context: ActorContext[SwapCommand]): ActorRef[Register.ForwardShortId[HasSwapId]] =
+    context.messageAdapter[Register.ForwardShortId[HasSwapId]](ForwardFailure)
+
+  def fundOpening(request: SwapInRequest, agreement: SwapInAgreement, invoice: Bolt11Invoice)(implicit context: ActorContext[SwapCommand], wallet: OnChainWallet, feeRatePerKw: FeeratePerKw): Unit = {
```

Comment:
I couldn't filed any explanation for how to use pilots to pass local fixed parameters, so I instead I just used currying to delineate them instead of using implicit. I still pass `context` as an implicit parameter though. see[4fcceb5](https://github.com/ACINQ/eclair/pull/2342/commits/4fcceb51723076fe756dfe310161c988bdf20630).

I could also define and use a local function like the following, but it seemed clearer to use the helper directly.
```
private def mySend(message: HasSwapId) = send(register, shortChannelId)(message)
```
Does this approach look ok?